ext/opcache/jit/zend_jit_ir.c

```diff
@@ -9691,7 +9691,7 @@ static int zend_jit_do_fcall(zend_jit_ctx *jit, const zend_op *opline, const zen
 		}
 	}
 
-	if (opline->opcode == ZEND_DO_FCALL) {
+	if (opline->opcode == ZEND_DO_FCALL || opline->opcode == ZEND_DO_UCALL) {
```

Comment:
Unrelated: Shouldn't this handle `ZEND_DO_FCALL_BY_NAME`?

Reply:
Apparently not?

```
$ sapi/cli/php -d zend_extension=(pwd)/modules/opcache.so -d opcache.enable_cli=1 -d opcache.opt_debug_level=0x20000 -d opcache.jit=tracing -d opcache.protect_memory=1 --repeat 2 test2.php
Executing for the first time...

$_main:
     ; (lines=4, args=0, vars=0, tmps=0)
     ; (after optimizer)
     ; php-src/test2.php:1-6
0000 INCLUDE_OR_EVAL (require) string("test.php")
0001 INIT_FCALL_BY_NAME 0 string("test")
0002 DO_FCALL_BY_NAME
0003 RETURN int(1)

$_main:
     ; (lines=1, args=0, vars=0, tmps=0)
     ; (after optimizer)
     ; php-src/test.php:1-7
0000 RETURN int(1)

test:
     ; (lines=2, args=0, vars=0, tmps=0)
     ; (after optimizer)
     ; php-src/test.php:4-6
0000 ECHO string("foo")
0001 RETURN null

Deprecated: Function test() is deprecated in php-src/test2.php on line 5
fooFinished execution, repeating...

Deprecated: Function test() is deprecated in php-src/test2.php on line 5
foo
```

Reply:
I believe with the default JIT config, this is not enough to trigger a trace. But I'd need to take a closer look. You can do `-dopcache.jit_debug=0x1ff000` to dump the traces.

Reply:
I needed to adjust this when changing the compilation of deprecated userland functions from UCALL to FCALL_BY_NAME.

To reproduce this standalone, I needed to add `-dopcache.jit_hot_func=1`.

Reply:
@TimWolla I think this path would already be reachable if the function were declared in a different file (or below the call).

Reply:
Okay, the change broke JIT. I pushed a follow-up that fixes the issue for me, but I totally don't understand what I am doing.